src/main/scala/chisel3/util/MemInterface.scala

```diff
@@ -0,0 +1,337 @@
+package chisel3.util
+
+import chisel3._
+
+import chisel3.internal.Builder
+import chisel3.experimental.SourceInfo
+import chisel3.internal.sourceinfo.{MemTransform, SourceInfoTransform}
+import scala.language.reflectiveCalls
+import scala.language.experimental.macros
+
+/** A bundle of signals representing a read memory port.
+  *
+  * @tparam tpe The data type of the memory port
+  * @param addrWidth The width of the address signal
+  */
+class MemRdPortInterface[T <: Data](tpe: T, addrWidth: Int) extends Bundle {
+  val addr = Input(UInt(addrWidth.W))
+  val enable = Input(Bool())
+  val readValue = Output(tpe)
+}
```

Comment:
This going to be very bikesheddy but I feel like we should spend the time to make sure we pick APIs we like here. We are doing a lot of mixing of abbreviations and not abbreviations so I think we should try to be self-consistent, at least within categories of things. Also, the word `Interface` is suuuuper overloaded and not particularly useful in this circumstance IMO but others may disagree.

As such, I will propose code changes for my preference on each of these bundles to get this bikeshedding party started.

```suggestion
class MemReadPort[T <: Data](tpe: T, addrWidth: Int) extends Bundle {
  val addr = Input(UInt(addrWidth.W))
  val en = Input(Bool())
  val data = Output(tpe)
}
```